src/Agent.Worker/Build/TrackingConfig.cs

```diff
@@ -60,13 +81,32 @@ public TrackingConfig()
             // Set the other properties.
             CollectionId = executionContext.Variables.System_CollectionId;
             DefinitionId = executionContext.Variables.System_DefinitionId;
-            HashKey = hashKey;
-            RepositoryUrl = repository.Url.AbsoluteUri;
-            RepositoryType = repository.Type;
+            RepositoryUrl = primaryRepository?.Url.AbsoluteUri;
+            RepositoryType = primaryRepository?.Type;
             System = BuildSystem;
             UpdateJobRunProperties(executionContext);
+
+            foreach (var repo in repositories)
+            {
+                RepositoryTrackingInfo.Add(new Build.RepositoryTrackingInfo
+                {
+                    Identifier = repo.Alias,
+                    RepositoryType = repo.Type,
+                    RepositoryUrl = repo.Url.AbsoluteUri,
+                    SourcesDirectory = Path.Combine(SourcesDirectory, RepositoryUtil.GetCloneDirectory(repo)),
+                });
+            }
+
+            // Now that we have all the repositories set up, we can compute the config hash
+            HashKey = TrackingConfigHashAlgorithm.ComputeHash(CollectionId, DefinitionId, RepositoryTrackingInfo);
         }
 
+        [JsonIgnore]
+        public string FileLocation { get; set; }
+
+        [JsonIgnore]
+        public bool WasConvertedFromOldFormat { get; set; }
```

Comment:
is this temporary? I worry that this will be a source of confusion later on. Maybe ConvertedFromFormatVersion()?

Reply:
It is not temporary as far as I know. I am not sure when the original file format parsing can be removed. Maybe we can work on eliminating the need for it soon, but right now there are a couple of places that fork based on whether the file started off in the original format. You are right tho, I can probably make this make more sense. 
Are you suggesting just a rename or are you implying something more with that name?

Reply:
I think I was implying something more. The current version would implicitly be '1.0' and the "new" version '2.0'. The object would be coded to match the latest version.  If we did that, then we would just need to know what version the object was deserialized from. Does that make sense? It is quite possible I am over complicating things.

Reply:
Got it. That makes sense. That's kind of what I was thinking when I read your comments but I wanted to make sure I understood. I actually almost did that and then backed off. I don't remember why. I will take a look and see if I can do that or just remove it altogether.

Reply:
I took a closer look at every use of that member and was able to remove it completely. Unfortunately, RM has its own tracking manager and tracking config. Luckily, these changes don't seem to affect them at all.